components/formats-api/src/loci/formats/FormatTools.java

```diff
@@ -1436,31 +1436,39 @@ public static Time getTime(Double value, String unit) {
   }
   
   public static Length getPhysicalSize(Double value, String unit) {
-    if (unit != null) {
-      try {
-        UnitsLength ul = UnitsLength.fromString(unit);
-        int ordinal = ul.ordinal();
-        Length returnLength = UnitsLength.create(value, ul);
-
-        if (returnLength.value().doubleValue() > Constants.EPSILON) {
-          return returnLength;
-        }
-        
-        // If the requested unit produces a value less than Constants.EPSILON then we switch to the next smallest unit possible
-        // Using UnitsLength.values().length - 2 as a boundary so as not to include Pixel and Reference Frame as convertible units
-        while (returnLength.value().doubleValue() < Constants.EPSILON && ordinal < (UnitsLength.values().length - 2)) { 
-          ordinal++;
-          ul = UnitsLength.values()[ordinal];
-          returnLength = UnitsLength.create(value, ul);
-        }
-        if (returnLength.value().doubleValue() > Constants.EPSILON) {
-          return returnLength;
+    if (value != null && value != 0) {
+      if (unit != null) {
+        try {
+          UnitsLength ul = UnitsLength.fromString(unit);
+          int ordinal = ul.ordinal();
+          Length returnLength = UnitsLength.create(value, ul);
+  
+          if (returnLength.value().doubleValue() > Constants.EPSILON && returnLength.value().doubleValue() < Double.POSITIVE_INFINITY) {
+            return returnLength;
+          }
+          
+          // If the requested unit produces a value less than Constants.EPSILON then we switch to the next smallest unit possible
+          // Using UnitsLength.values().length - 2 as a boundary so as not to include Pixel and Reference Frame as convertible units
+          while (returnLength.value().doubleValue() < Constants.EPSILON && ordinal < (UnitsLength.values().length - 2)) { 
```

Comment:
Several `NullPointerException` seem to be thrown by this line -see https://ci.openmicroscopy.org/view/Failing/job/BIOFORMATS-5.1-merge-openbytes-performance/687/testngreports/loci.tests.testng/OpenBytesPerformanceTest/setId_101/ or ttps://ci.openmicroscopy.org/job/BIOFORMATS-5.1-merge-full-repository/